Maintenance: Remove Red Hat Linux workarounds predating RHEL

kinkie opened the pull request:
The last Red Hat Linux release went EOL in 2004, replaced by Red Hat
Enterprise Linux and Fedora Linux. We no longer support Red Hat Linux
releases and expect that these hacks are no longer necessary in
supported environments.

rousskov commented on configure.ac:
```diff
@@ -1158,12 +1158,6 @@ AS_IF([test "x$with_openssl" = "xyes"],[
     ],$LIBOPENSSL_LIBS)
   ])
 
-  # This is a workaround for RedHat 9 brain damage..
```
RedHat 9 was released on 2022-05-17. I assume this 2003 comment has a RedHat version typo.

I am curious how do you know that these RedHat workarounds are no longer needed (on still supported RedHat flavors).

yadij replied:
Original `RedHat 9`, not part of the `Enterprise Linux` series.

kinkie commented:
The RedHat 9 brain damage comment appears as context in commit c41db002c1b0e10b9b753d092474f752bd1ded2f (2014), the one about setresuid brokenness as context in commit 14c0d3ab6724310f2e7bff906b65f7ddd88df058 (2012).

These hacks are older than that, so no, I don't think it's a typo.

We know they are not longer needed if our CI build succeds for this PR - it tries every supported redhat release

rousskov commented:
> The RedHat 9 brain damage comment appears as context in commit [c41db00](https://github.com/squid-cache/squid/commit/c41db002c1b0e10b9b753d092474f752bd1ded2f) (2014)

That comment and the corresponding hack appears in 2003 commit f11555e.


> These hacks are older than that, so no, I don't think it's a typo.

I do not understand why you think it is not a typo: How can a 2003 comment/hack/commit refer to RedHat 9 that was released in 2022 (unless that reference was a typo)?

> We know they are not longer needed if our CI build succeds for this PR - it tries every supported redhat release

What is the earliest RedHat release that our CI tests?

kinkie commented:
> > The RedHat 9 brain damage comment appears as context in commit [c41db00](https://github.com/squid-cache/squid/commit/c41db002c1b0e10b9b753d092474f752bd1ded2f) (2014)
> 
> 
> 
> That comment and the corresponding hack appears in 2003 commit f11555e.
> 
> 
> 
> 
> 
> > These hacks are older than that, so no, I don't think it's a typo.
> 
> 
> 
> I do not understand why you think it is not a typo: How can a 2003 comment/hack/commit refer to RedHat 9 that was released in 2022 (unless that reference was a typo)?
> 
> 
> 
> > We know they are not longer needed if our CI build succeds for this PR - it tries every supported redhat release
> 
> 
> 
> What is the earliest RedHat release that our CI tests?

Aah, it's now clear.
These comments do not refer to Red Hat Enterprise Linux 9, but to Red Hat Linux 9.0, which was discontinued in 2004 https://en.m.wikipedia.org/wiki/Red_Hat_Linux#:~:text=Red%20Hat%20Linux%20was%20a,on%20Red%20Hat%20Linux%209

yadij commented:
Now that the meaning of the comment has been sorted out, do we have evidence that all of the current `RHEL` versions have resolved this issue?

FYI, there are enough RHEL-downstream distros still shipping latest Squid on top of old `RHEL 7.10` bases that we will have to include that in the decision to drop this hack.

rousskov commented:
> Alex: What is the earliest RedHat release that our CI tests?

> Amos: do we have evidence that all of the current RHEL versions have resolved this issue?

I added "[we] hope that these hacks are not needed on RHEL" to the PR description. Once/if we get the above info, we can replace that phrase with something better.

kinkie commented:
> > Alex: What is the earliest RedHat release that our CI tests?
> 
> > Amos: do we have evidence that all of the current RHEL versions have resolved this issue?
> 
> I added "[we] hope that these hacks are not needed on RHEL" to the PR description. Once/if we get the above info, we can replace that phrase with something better.

Current [support cycles by red hat](https://access.redhat.com/product-life-cycles?product=Red%20Hat%20Enterprise%20Linux,OpenShift%20Container%20Platform%204):
RHEL 7 is in "maintenance support" until June 2024 ; it was not tested
RHEL 8 is fully supported, it was tested (both as CentOS 8 and CentOS stream 8), no problems
RHEL 9 is fully supported, it was tested (against the CentOS stream variant), no problems

Red Hat does not make RHEL images or containers available, Centos 9 is not made available either

kinkie commented:
Reworded slightly, hopefully this is good for all to land

rousskov reviewed (approved):
> Reworded slightly, hopefully this is good for all to land

I fixed punctuation and added "in supported environments" (without detailing what those are -- I am still unsure what those are).

rousskov commented:
> I fixed punctuation and added "in supported environments" (without detailing what those are -- I am still unsure what those are).

I did not want these PR description changes to merge without others having a chance to look at them, so I removed the M-cleared-for-merge label from this unanimously approved PR. Please add it back when you review the updated PR description.

kinkie commented:
> > I fixed punctuation and added "in supported environments" (without detailing what those are -- I am still unsure what those are).
> 
> I did not want these PR description changes to merge without others having a chance to look at them, so I removed the M-cleared-for-merge label from this unanimously approved PR. Please add it back when you review the updated PR description.

Works for me, readding. Thanks!
